pkg/wireguard/agent/agent.go

```diff
@@ -15,6 +15,10 @@ import (
 	"os"
 	"strconv"
 
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
```

Comment:
hmmm, i think that this generates the new dependency for k8s client `InClusterConfig`. as far as i can see that `wireguard` agent is NOT dependent on k8s but cilium only. i guess that this is designed to avoid?

Reply:
as i mentioned before, this should be implemented in `node` package, and `wireguard` agent should process as it requested by `node` manager?

Reply:
my fault, I miss-understand the previous comments, I think implement in `node` package is better. I miss-understand the previous comments as `node-handle.go` files.
I will check this and try to modify the code